application/helpers/SurveyRuntimeHelper.php

```diff
@@ -314,7 +314,12 @@ public function run($surveyid, $args)
                 $aGroup['class'] = ' ls-hidden';
             }
 
-            $aGroup['name']        = $gl['group_name'];
+            // Set basic replacements to use on group texts
+            $aStandardsReplacementFields = getStandardsReplacementFields($this->aSurveyInfo);
+            $aStandardsReplacementFields['GID'] = $gid;
+            $aStandardsReplacementFields['GROUPNAME'] = $gl['group_name'];
+
+            $aGroup['name']        = LimeExpressionManager::ProcessString($gl['group_name'], null, $aStandardsReplacementFields, 3, 1);
```

Comment:
Strip {GROUPNAME} to avoid recursion.

Reply:
Wouldn;t that happen with any other field replacement?

Reply:
On this new commit, we are using, the processed group name to use a replacement value, and not the original expression. Is that what you meant?